File: src/main/resources/static/js/service/goodsTypeService.js
//服务层
app.service('goodsTypeService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../type/findAll');
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../type/findPage?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../type/findOne?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../type/add',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../type/update',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../type/delete?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../type/search?page='+page+"&rows="+rows, searchEntity);
	}

	//查询子分类
	this.selectByPId=function(PId,page,rows){
		return $http.get('../type/selectByPId?page='+page+'&rows='+rows+'&PId='+PId);
	}
});

File: src/main/java/cdtu/wheretobuy/service/GoodsTypeService.java
package cdtu.wheretobuy.service;
import java.util.List;
import cdtu.wheretobuy.pojo.GoodsType;

import cdtu.wheretobuy.pojo.PageResult;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface GoodsTypeService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<GoodsType> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(GoodsType type);
	
	
	/**
	 * 修改
	 */
	public void update(GoodsType type);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public GoodsType findOne(Integer id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Integer[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(GoodsType type, int pageNum, int pageSize);

	/**
	 * 查询子分类
	 * @param PId 父id
	 * @return
	 */
	public PageResult selectByPId(Integer PId,int pageNum, int pageSize);
	
}
